### PR TITLE
Modnomod in watch

### LINKED ISSUE #721: Option to use esm build with dev server
**Do you want to request a *feature* or report a *bug*?**
feature

**What is the current behaviour?**
no option to use esm build with dev server

**If the current behaviour is a bug, please provide the steps to reproduce.**

**What is the expected behaviour?**
--esm flag for dev server (preact watch)

**If this is a feature request, what is motivation or use case for changing the behaviour?**
we have been noticing issues that show up only in esm builds and would like to be able to fix them while using preact watch for eg. this fails in esm build only -
```
render({
		image = requireGameImage(`${gameId}/eventTile.jpg`),
		game: gameName,
		gameId
	 }.....
```

**Please mention other relevant information.**
- Node version latest
- npm version latest
- Operating system windows
- CLI version 3.0.0-next.16
- Browser chrome

### PR DESCRIPTION
**Pull request by prateekbh**
Fixes #721 

**What kind of change does this PR introduce?**

Adds module/nomodule in `preact watch`

**Did you add tests for your changes?**
No

**Summary**

Adds the esm capability in `preact watch`.

**Does this PR introduce a breaking change?**

No

### CODE REVIEW

File: packages/cli/lib/lib/webpack/webpack-client-config.js [hunk 1]
```diff
@@ -180,44 +231,6 @@ function isProd(config) {
 	}
 
 	if (config.esm) {
```
Comment (ForsakenHarmony) [inline packages/cli/lib/lib/webpack/webpack-client-config.js]: guess this can be simplified now

Comment (prateekbh) [reply packages/cli/lib/lib/webpack/webpack-client-config.js]: fixed

File: packages/cli/lib/lib/webpack/webpack-client-config.js [hunk 2]
```diff
@@ -109,6 +113,53 @@ function clientConfig(env) {
 	};
 }
 
+function getBabelEsmPlugin(config) {
```
Comment (ForsakenHarmony) [inline packages/cli/lib/lib/webpack/webpack-client-config.js]: why did you move this out btw?

Comment (prateekbh) [reply packages/cli/lib/lib/webpack/webpack-client-config.js]: Oh right.. This was intermittent decision which should have not been moved out

Comment (ForsakenHarmony) [reply packages/cli/lib/lib/webpack/webpack-client-config.js]: oh 👀
